app/models/zipcode.rb

```diff
@@ -0,0 +1,35 @@
+# Generated by "bin/rails g my_zipcode_gem:models", lightly modified to make rubocop happy
+class Zipcode < ApplicationRecord
+  belongs_to :county
+  belongs_to :state
+
+  validates :code, uniqueness: true, presence: true
+  validates :state_id, :county_id, :city, presence: true
+
+  scope :without_county, -> { where('county_id IS NULL') }
+  scope :without_state, -> { where('state_id IS NULL') }
+  scope :ungeocoded, -> { where('lat IS NULL OR lon IS NULL') }
+
+  class << self
+    def find_by_city_state(city, state)
+      includes(county: :state)
+        .where('city like ? AND states.abbr like ?', "#{city}%", "%#{state}%")
+        .references(:state)
+        .first
+    end
+
+    def find_all_by_city_state(city, state)
+      includes(county: :state)
+        .where('city like ? AND states.abbr like ?', "#{city}%", "%#{state}%")
+        .references(:state)
+    end
+  end
```

Comment:
Elsewhere in the app, we tend to do `def self.class_method_name_here` instead of using the indented `class << self` blocks. I tend to think the latter becomes difficult to read and less-obvious that methods are class methods vs. instance methods.